### DISCUSSION TITLE
Unable make a new clean project works with `laravel-permission`

### DISCUSSION THREAD

**Discussion by realtebo** [Laravel-Backpack/community-forum#566] [Q&A (Help)] [answered]
I installed `laravel-permission` package and executed all migrations

**I didn't touch any config about auth / middleware / etc.**

I am creating a role and a permission, giving the permission to the role.

```
app()[\Spatie\Permission\PermissionRegistrar::class]->forgetCachedPermissions();
$sysadmin_role = Role::create(['name' => 'sysadmin']);
$permission = Permission::create(['name' => 'see-sysadmin-panels']);
$sysadmin_role->givePermissionTo($permission);
app()[\Spatie\Permission\PermissionRegistrar::class]->forgetCachedPermissions();
```

This works, I can see in db all is created, all using `web` guard.

I am creating an user and giving it the role 'sysadmin'

```
$sysadmin = User::factory()->create([
           'email' => 'sysadmin@test.it',
           'name' => 'SysAdmin',
]);
app()[\Spatie\Permission\PermissionRegistrar::class]->forgetCachedPermissions();

$sysadmin->assignRole('sysadmin');

// Reset cached roles and permissions
app()[\Spatie\Permission\PermissionRegistrar::class]->forgetCachedPermissions();
```

This works, I can see in db all relations configured

```
λ php artisan permission:show
Guard: web
+---------------------+---------+-------+---------+----------+---------+
|                     | manager | staff | student | sysadmin | teacher |
+---------------------+---------+-------+---------+----------+---------+
| see-sysadmin-panels |  ·      |  ·    |  ·      |  ✔       |  ·      |
+---------------------+---------+-------+---------+----------+---------+
```

then i  a blade i am using

```
@can('see-sysadmin-panels')
    ONLY VISIBLE IF SEE SYSADMIN PANELS
@endcan
```

The problem: When loggedin as sysadmin, I cannot see the above text

Why ? What should I do in a plain new backpack 6 project to integrate with `laravel-permission`?

**Comment by realtebo** [comment, marked as answer]
Find !

The solution is in `config/backpack/base.php` to change row 

    'guard' => 'backpack'
  
into

    'guard' => config('auth.defaults.guard', 'backpack')

**Reply by tabacitu** [reply]
Thanks for sharing the solution @realtebo

**Comment by realtebo** [comment]
Or ... better ... leave backpack untouched and when creating permissions and so on set 'backpack' as guard
